### PR TITLE
Feature/improved api calls

### LINKED ISSUE #66: Finish API Calls
- [ ] Add code to handle errors for loading the survey and for submitting the survey
- [ ] Submit subscription to back end when a user finished the survey. This should use the `create` action in the `subscriptions` directory.

### PR DESCRIPTION
**Pull request by deadbelly**
### Fix/Feature/Bug?
Feature, but with a few fixes along the way.

### What does this PR do?
Mostly this branch improves and polishes our api calls. I've added the call that creates a new subscription back in, but this time at the end of the survey, and I've added error handling to that call, and to the call that starts the survey. The biggest change that makes is all run smoothly is adding a `crateId` property to the `survey` state, letting us send that int along to the server when it's time to submit.

While testing I noticed several small bugs/issues that I fixed along the way. These included: missing assets, old console logs, old commented code, a lack of prop typechecking, and the keys for edgy and classy styles being switched (sorry!). I fixed all these issues so the app should run a bit more smoothly now and the code should look cleaner. Tomorrow post-code freeze we can look at it in more detail and see if it needs more cleanup. 

### Where should review start?
Look at `survey/survey.js` to see how adding a subscription is handled, and `subscriptions/api/actions.js` for the api call. Check `crate/item.js` for the error handling on launching the survey. 

### How is this tested?
Run through the user stories that involve adding a subscription and watch it succeed! For sad path testing stubbed calls in cypress might be best. I'm willing to add these or help add these tonight or tomorrow. When working with this code make sure to re-seed your database and also LOG OUT and then log back in. 

`npx sequelize-cli db:seed:undo:all`

`npx sequelize-cli db:seed:all`

### Applicable tickets?
closes #66

### CODE REVIEW

**Comment by deadbelly** [comment]
I just wanted to note that I just rebased this branch and force pushed the new version. Kelsie and my branches went up around the same time and hers was more clearly ready to merge since it was all new files and mine changes old files, so I merged it first.

File: code/web/src/modules/survey/Survey.js [hunk 1]
```diff
@@ -95,10 +90,32 @@ class Survey extends PureComponent {
...
-          
+
+          this.props.create({
+            style: this.getResults(),
+            crateId: this.props.survey.crateId
+          }).then(response => {
+              if (response.data.errors && response.data.errors.length > 0) {
+                this.props.messageShow(response.data.errors[0].message)
+              } else {
+                this.props.messageShow('Subscribed successfully.')
```
Comment (DrewBradley) [inline code/web/src/modules/survey/Survey.js:101]: Thanks for making this show the right message!!!

File: code/web/src/modules/survey/Survey.js [hunk 2]
```diff
@@ -131,17 +148,28 @@ class Survey extends PureComponent {
...
+  previousPage: PropTypes.func.isRequired,
+  getProducts: PropTypes.func.isRequired,
+  clearSurvey: PropTypes.func.isRequired,
+  toggleSelection: PropTypes.func.isRequired,
+  retakeSurvey: PropTypes.func.isRequired,
+  create: PropTypes.func.isRequired,
+  messageShow: PropTypes.func.isRequired,
+  messageHide: PropTypes.func.isRequired,
+  survey: PropTypes.object.isRequired
+}
```
Comment (DrewBradley) [inline code/web/src/modules/survey/Survey.js:164]: Good catch! Mad props!

File: code/web/src/modules/survey/api/actions.js [hunk 3]
```diff
@@ -77,7 +78,8 @@ export const clearSurvey = () => {
     dispatch({
       type: 'SURVEY_CLEAR',
       products: {},
-      page: 1
+      page: 1,
+      crateId: 0
```
Comment (DrewBradley) [inline code/web/src/modules/survey/api/actions.js:82]: Nicely done!

File: code/web/src/setup/helpers.js [hunk 4]
```diff
@@ -71,8 +71,8 @@ export function formatSurvey(products) {
 export function styleToString(style) {
   const key = {
     1: 'Casual',
-    2: 'Edgy',
-    3: 'Classy'
+    2: 'Classy',
+    3: 'Edgy'
```
Comment (DrewBradley) [inline code/web/src/setup/helpers.js:75]: I was wondering about this, but I wasn't sure. I guess I'm actually 'Classy' after all.

**Review by DrewBradley — APPROVED** [review]
Thank you for cleaning up that API call and adding propTypes!

**Pull request merged**